android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt

```diff
@@ -101,6 +100,22 @@ constructor(
     configurationRegistry.retrieveConfiguration(ConfigType.Navigation)
   }
 
+  fun retrieveRemoteIconsBitmap() {
+    navigationConfiguration.clientRegisters
+      .filterNot { it.menuIconConfig == null }
+      .filter { it.menuIconConfig!!.type == "remote" }
+      .forEach {
+        val resourceId = it.menuIconConfig?.reference!!.substringAfter("/")
+        viewModelScope.launch {
```

Comment:
This requires a dispatcher context like `io()`  otherwise the `main` context will be used.

Reply:
That's true, will add that